packages/picasso-forms/src/Form/mutators/set-active-field-touched.ts

```
@@ -8,10 +8,9 @@ export const setActiveFieldTouched = <
   state: MutableState<FormValues, InitialFormValues>
 ) => {
   const activeFieldName = state.formState.active
+  const field = activeFieldName && state.fields[activeFieldName]
 
-  if (activeFieldName) {
-    const field = state.fields[activeFieldName]
-
+  if (field) {
```

Slightly refactored to avoid nesting

---

Did we reach the field as undefined before? The question is more like, should we fail silently like here and don't apply `touched`, or should we throw an early error so people can fix the root problem? Just some food for thought, as I don't know the full context of the problem :thinking:

---

The full context is here https://toptal-core.slack.com/archives/CERF5NHT3/p1692093998693089?thread_ts=1691993118.653259&cid=CERF5NHT3. Problem is not reproducible but cause application to fail. This fix will make it not fail at least in this function. The error has been thrown and generated Sentry report :)

---

Got it, makes sense to guard it, but we need to check if an error will not show up in other places after this fix. Maybe the root of the problem is somewhere else, and not having a field here is a symptom of that problem

---

I am not sure we will be able to check as even app owners can't reproduce the problem. I believe this error will reappear in some other form in the app and then app maintainers will be able to fix it. IMO it is unlikely the error is in Picasso. Picasso just failed because of some abuse (I am only hypothesising).